### ISSUE TITLE
Why do we need to construct `one-hot` vectors of the input_ids and multiple by the embeddings as opposed to applying the embedding directly?

### ISSUE THREAD

**Issue by vgoklani** [jalammar/ecco#43]
Hey there,

Thanks for releasing this library! I was reviewing your lm.py file, and in particular, I was unclear why you were constructing one-hot vectors and multiplying by the embedding matrix, as opposed to simply applying the embedding directly.

See here:

https://github.com/jalammar/ecco/blob/main/src/ecco/lm.py#L118

my approach:

	import torch
	from transformers import pipeline

	# https://huggingface.co/transformers/quicktour.html
	classifier = pipeline('sentiment-analysis', model="distilbert-base-uncased-finetuned-sst-2-english")

	model = classifier.model
	tokenizer = classifier.tokenizer

	text = "We are very happy to show you the 🤗 Transformers library."

	encoding = tokenizer.encode_plus(
		text,
		return_tensors="pt",
		add_special_tokens=True,
		return_attention_mask=True,
	)

	inputs_embeds = model.base_model.embeddings(encoding['input_ids'])

	assert inputs_embeds.is_leaf is False
	inputs_embeds.retain_grad()

	logits = model(inputs_embeds=inputs_embeds, attention_mask=encoding['attention_mask']).logits.squeeze(dim=0)

	score = logits[logits.argmax()]
	score.backward(gradient=None, retain_graph=True)

	inputs_embeds__grad = (inputs_embeds.grad * inputs_embeds)[:, 1:-1, :]  # remove CLS and SEP tokens

	feature_importance = torch.norm(inputs_embeds__grad, dim=2)
	feature_importance_normalized = (feature_importance / torch.sum(feature_importance)).squeeze(dim=0)

	attributions = [{tokenizer.convert_ids_to_tokens(input_id.item()): feature_importance_normalized[index].item()} for index, input_id in enumerate(encoding['input_ids'].squeeze(dim=0)[1:-1])]

note how the inputs_embeds are calculated directly.
